src/data_insertion.py

```diff
@@ -1,188 +1,131 @@
-from src.logger import LOGGER
-from src.config import *
-from src.utils import connect_to_pinecone_index, split_text
-from tqdm.auto import tqdm
-import time
 import json
+import time
+
 import openai
+from langchain.embeddings.openai import OpenAIEmbeddings
+from tqdm.auto import tqdm
+
+from src.config import ES_INDEX, OPENAI_API_KEY, OPENAI_EMBED_MODEL
+from src.logger import LOGGER
+
+
 openai.api_key = OPENAI_API_KEY
 
 
-def individual_data_insertion(pinecone_index, embed, data):
-    url = data['url']
-    if data['url'] is None:
-        url = "No url in data"
-        
-    text = data['clean_text']
-    if isinstance(text, list):
-        text = " ".join(text)
-    # we shorten the text and give link if text size exceed
-    if len(text.encode('utf-8')) > 30000:
-        text = split_text(text, 30)[0] + f"...for more details you can click on below link\n {data['url']}"
-    
-    id = [data['es_id']]
-
-    domain = data['domain']
-    if data['domain'] is None:
-        domain = ""
-
-    created_at = data['created_at']
-    if data['created_at'] is None:
-        created_at = ""
-
-    metadata_id = data['id']
-    if data['id'] is None:
-        metadata_id = ""
-    
-    title = data['title']
-    if data['title'] is None:
-        title = ""
-    
-    type_ = data['type']
-    if data['type'] is None:
-        type_ = ""
-    
-    authors = data['authors']
-    if data['authors'] is None:
-        authors = ""
-
-
-    meta_data = [{
-        'domain': domain,
-        'created_at': created_at,
-        'id': metadata_id,
-        'title': title,
-        'text': text,
-        'type': type_,
-        'url': url,
-        'authors': authors
-    }]
-
-    to_upsert = list(zip(id, embed, meta_data))
-
-    # connect and upsert to Pinecone
-    pinecone_index.upsert(vectors=to_upsert)
-
-
-def data_embed_insertion(pinecone_index, data, start_from):
-    LOGGER.info("Generating of embeddings of text and insertion of data started...")
+def extract_metadata(data):
+    # TODO:- Add `indexed_at` field
+    url = data["url"] if data["url"] else "No url in data"
```

Comment:
why `No url in data` rather than None? Does not make more sense to use None so we have null values in data?

Reply:
I didn't think much about this, just copied whatever the code did before. But I concur with your suggestions. We should store `None` instead of some empty or arbitrary string.